### PR TITLE
Process headers to allow shibb and ip authentication

### PR DESCRIPTION
**Pull request by terrywbrady**
See https://jira.duraspace.org/browse/DS-2898

I have tested this code with Shibb authentication and with IP authentication.  The headers are processed to pull Shibb credentials or IP credentials.

A call was also added to process SpecialGroup permissions.

Call /login-shibboleth to retrieve a Shibboleth login page from the authentication manager.  That code redirects to /shibboleth-login after login.  I created a method to present the user's authentication credentials after login.  I presume that this needs to be cleaned up a bit, and I am seeking suggestions on this code.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java [hunk 1]
```diff
@@ -226,7 +226,12 @@ public int authenticate(Context context, String username, String password,
 			// Step 4: Log the user in.
 			context.setCurrentUser(eperson);
 			request.getSession().setAttribute("shib.authenticated", true);
-            authenticationService.initEPerson(context, request, eperson);
+
+			//When testing, this object did not seem to be properly initialized on the first test
+			if (authenticationService == null) {
+				authenticationService = AuthenticateServiceFactory.getInstance().getAuthenticationService();
```
Comment (terrywbrady) [inline dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java]: This variable was not properly initialized when this code was invoked.  Is there a cleaner way to ensure that the proper objects have been initialized?

**Comment by KevinVdV** [comment]
Currently -1 on this pull request as we might want to rethink our rest authentication as whole, see comment made in: https://jira.duraspace.org/browse/DS-2898

**Comment by terrywbrady** [comment]
@KevinVdV , your suggestions on password authentication sound like a new issue.  Perhaps an additional Jira ticket should be submitted to capture those concerns..

The intention of this PR is to ensure that sessions authenticated by Shibboleth and by IP are able to pass in their credentials.

Regardless of the enhancements that are made to the password authentication process, I think some form of these changes will still be needed.

**Comment by KevinVdV** [comment]
Alternate pull request: https://github.com/DSpace/DSpace/pull/1226

This pull request solves the issue I have with the current authentication while also supporting shibboleth (or so I hope, since I have no way of testing).

**Comment by tdonohue** [comment]
Closed in favor of #1226

**Pull request closed without merging**